packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts

```diff
@@ -203,8 +186,53 @@ class MedicalDataService {
     return this._datumOpts.timezoneName
   }
 
-  add(rawData: Array<Record<string, unknown>>): void {
-    this.normalize(rawData)
+  add(data: MedicalData): void {
+    if (data.bolus) {
+      this.medicalData.bolus = this.medicalData.bolus.concat(data.bolus)
+    }
+    if (data.basal) {
+      this.medicalData.basal = this.medicalData.basal.concat(data.basal)
+    }
+    if (data.cbg) {
+      this.medicalData.cbg = this.medicalData.cbg.concat(data.cbg)
+    }
+    if (data.confidentialModes) {
+      this.medicalData.confidentialModes = this.medicalData.confidentialModes.concat(data.confidentialModes)
+    }
+    if (data.deviceParametersChanges) {
+      this.medicalData.deviceParametersChanges = this.medicalData.deviceParametersChanges.concat(data.deviceParametersChanges)
+    }
+    if (data.messages) {
+      this.medicalData.messages = this.medicalData.messages.concat(data.messages)
+    }
+    if (data.meals) {
+      this.medicalData.meals = this.medicalData.meals.concat(data.meals)
+    }
+    if (data.physicalActivities) {
+      this.medicalData.physicalActivities = this.medicalData.physicalActivities.concat(data.physicalActivities)
+    }
+    if (data.pumpSettings) {
+      this.medicalData.pumpSettings = this.medicalData.pumpSettings.concat(data.pumpSettings)
+    }
+    if (data.reservoirChanges) {
+      this.medicalData.reservoirChanges = this.medicalData.reservoirChanges.concat(data.reservoirChanges)
+    }
+    if (data.smbg) {
+      this.medicalData.smbg = this.medicalData.smbg.concat(data.smbg)
+    }
+    if (data.warmUps) {
+      this.medicalData.warmUps = this.medicalData.warmUps.concat(data.warmUps)
+    }
+    if (data.wizards) {
+      this.medicalData.wizards = this.medicalData.wizards.concat(data.wizards)
+    }
+    if (data.zenModes) {
+      this.medicalData.zenModes = this.medicalData.zenModes.concat(data.zenModes)
+    }
+    if (data.alarmEvents) {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏So sad to have all `if()` conditions sorted in alphabetical order, to find `data.alarmEvents` at the end... :joy:

Reply:
ok